packages/authorization-process/src/services/authorizationService.ts

```diff
@@ -565,9 +576,88 @@ export function authorizationServiceBuilder(
         )
       );
     },
+
+    async createKeys({
+      clientId,
+      authData,
+      keysSeeds,
+      correlationId,
+      logger,
+    }: {
+      clientId: ClientId;
+      authData: AuthData;
+      keysSeeds: ApiKeysSeed;
+      correlationId: string;
+      logger: Logger;
+    }): Promise<{ client: Client; showUsers: boolean }> {
+      logger.info(`Creating keys for client ${clientId}`);
+      const client = await retrieveClient(clientId, readModelService);
+      assertOrganizationIsClientConsumer(
+        unsafeBrandId(authData.organizationId),
+        client.data
+      );
+      assertKeyIsBelowThreshold(
+        clientId,
+        client.data.keys.length + keysSeeds.length
+      );
+      if (!client.data.users.includes(authData.userId)) {
+        throw userNotFound(authData.userId, authData.selfcareId);
+      }
+
+      await assertUserSelfcareSecurityPrivileges(
+        authData.selfcareId,
+        authData.userId,
+        authData.organizationId,
+        selfcareV2Client,
+        client.data.users.find((userId) => userId === authData.userId)
+      );
+
+      if (keysSeeds.length !== 1) {
+        throw genericInternalError("Wrong number of keys"); // TODO should we add a specific error?
+      }
+      const keySeed = keysSeeds[0];
+      const jwk = createJWK(decodeBase64ToPem(keySeed.key));
+      const newKey: Key = {
+        clientId,
+        name: keySeed.name,
+        createdAt: new Date(),
+        kid: calculateKid(jwk),
+        encodedPem: keySeed.key,
+        algorithm: keySeed.alg,
+        use: ApiKeyUseToKeyUse(keySeed.use),
+        userId: authData.userId,
+      };
+      const duplicateKid = await readModelService.getKeyByKid(newKey.kid);
+      if (duplicateKid) {
+        throw keyAlreadyExists(newKey.kid);
+      }
+      const updatedClient: Client = {
+        ...client.data,
+        keys: [...client.data.keys, newKey],
+      };
+      await repository.createEvent(
+        toCreateEventKeyAdded(
+          newKey.kid,
+          updatedClient,
+          client.metadata.version,
+          correlationId
+        )
+      );
+
+      return {
+        client: updatedClient,
+        showUsers: true,
+      };
+    },
   };
 }
 
 export type AuthorizationService = ReturnType<
   typeof authorizationServiceBuilder
 >;
+
+const assertKeyIsBelowThreshold = (clientId: ClientId, size: number): void => {
```

Comment:
What about `assertKeysCountIsBelowThreshold`?

Reply:
As you prefer